Runtimes/Core/runtime/CMakeLists.txt

```diff
@@ -86,7 +95,8 @@ target_compile_definitions(swiftRuntime
     $<$<BOOL:${SwiftCore_ENABLE_STDLIB_TRACING}>:-DSWIFT_STDLIB_TRACING>
     $<$<BOOL:${SwiftCore_ENABLE_SHORT_MANGLING_LOOKUPS}>:-DSWIFT_STDLIB_SHORT_MANGLING_LOOKUPS>
     $<$<BOOL:${SwiftCore_HAS_DLADDR}>:-DSWIFT_STDLIB_HAS_DLADDR>
-    $<$<BOOL:${SwiftCore_HAS_DLSYM}>:-DSWIFT_STDLIB_HAS_DLSYM>)
+    $<$<BOOL:${SwiftCore_HAS_DLSYM}>:-DSWIFT_STDLIB_HAS_DLSYM>
+    $<$<BOOL:${SwiftCore_ENABLE_FILESYSTEM_SUPPORT}>:-DSWIFT_STDLIB_HAS_FILESYSTEM$<SEMICOLON>-DSWIFT_ARCH="${SwiftCore_ARCH_SUBDIR}"$<SEMICOLON>-DSWIFT_LIB_SUBDIR="${SwiftCore_PLATFORM_SUBDIR}">)
```

Comment:
Why not break this up? It would replicate the condition, but that seems easier to read.

Reply:
Was trying to indicate that they were all related. I can split them up if you prefer though.